video/out/opengl/shader_cache.c

```diff
@@ -274,9 +274,28 @@ void gl_sc_uniform_image2D_wo(struct gl_shader_cache *sc, const char *name,
     u->v.tex = tex;
 }
 
+void gl_sc_ubo(struct gl_shader_cache *sc, char *name, struct ra_buf *buf,
+               char *format, ...)
```

Comment:
Better to make it take only a string, and have the caller use mp_tprintf if formatting is necessary?

Reply:
I figured it'd be easier if gl_sc is directly in charge of memory management of the formatted string, saves an allocation/free. Also, I think since gl_sc_ is supposed to be a “nicer” front-end to use directly in opengl/video.c, so I prefer to go with the API that's easier to use.

Btw, I was also thinking about getting rid of gl_sc_add() in favor of just gl_sc_addf(), and making GLSL and GLSLF the same macro (just “GLSL”). Thoughts?

Reply:
mp_tprintf doesn't require a free.

Reply:
Oh, that's the constant sized version. I see what you mean. Your call.

Reply:
My call is reducing ... functions.